lib/helpers.js

```diff
@@ -60,11 +60,13 @@ module.exports.getFirstWord = function (message) {
 };
 
 module.exports.isYesResponse = function (message) {
-  const trimmed = this.getFirstWord(message.toLowerCase().trim());
+  const trimmedMsg = message.toLowerCase().trim();
   const yesResponses = process.env.GAMBIT_YES_RESPONSES ?
     process.env.GAMBIT_YES_RESPONSES.toLowerCase().split(',') : ['yes'];
 
-  return yesResponses.indexOf(trimmed) >= 0;
+  const matchRegex = new RegExp(`^(${yesResponses.join('|')})$`);
```

Comment:
The change attempts to behave as close as possible to the original business logic (which was to only contemplate the beginning of the text) while fixing the bug. In this case "I'm thinking about it and the answer is YES" is not a Yes response now, or before the code changes. If the business logic has changed, we should make a separate issue and address that.